webplugin/js/app/km-widget-custom-events.js

```diff
@@ -0,0 +1,196 @@
+/**
+ // Kommunicate custom events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+var kmWidgetEvents = {
+    sendEventToGoogleAnalytics: function (
+        eventCateogry,
+        eventAction,
+        eventLabel,
+        eventValue
+    ) {
+        var trackingID = applozic._globals.gaTrackingID;
+        if (trackingID && typeof window.top.ga !== 'undefined') {
+            window.top.ga('create', trackingID.toString(), 'auto');
+            window.top.ga('send', {
+                hitType: 'event',
+                eventCategory: eventCateogry,
+                eventAction: eventAction,
+                eventLabel: eventLabel,
+                eventValue: eventValue,
+            });
+        }
+    },
+    eventTracking: function (eventObject, customLabel, customValue) {
+        // Any other analytics tool related code can be add here no need to paste it in every function
+        if (kommunicateCommons.isObject(eventObject)) {
+            if (customLabel) {
+                eventObject.data.eventLabel = customLabel;
+            }
+            if (customValue) {
+                eventObject.data.eventValue = customValue;
+            }
+            applozic._globals.gaTrackingID &&
+                kmWidgetEvents.sendEventToGoogleAnalytics(
+                    eventObject.eventCateogry,
+                    eventObject.eventAction,
+                    eventObject.eventLabel,
+                    eventObject.eventValue
+                );
+            if (eventObject.eventFunction !== null) {
+                // checks if there is any errors in user provided function
+                try {
+                    eventObject.eventFunction(eventObject.data);
+                } catch (error) {
+                    console.log(error.stack);
+                }
+            }
+        } else {
+            throw new TypeError(
+                'eventTracking expect an object but got ' + typeof eventObject
```

Comment:
if this error is thrown, what will happen?
will it fail the chat widget functionality?

Reply:
corrected